cmd/device-worker/main.go

```diff
@@ -230,6 +231,9 @@ func main() {
 	go listenStartGracefulRebootChannel(wl, dataMonitor, systemMetricsWatcher, metricsStore, hbs, ansibleManager,
 		gracefulRebootChannel, deviceOs)
 
+	// Start prometheus httpd handler
+	go datatransfer.StartMetrics()
```

Comment:
I do not see a reason to start a new goroutine and an HTTP server for collecting our own metrics.

Reply:
This routine is to expose the new metrics, not to collect them.

Reply:
I see your point, but as I mentioned later on in @jakub-dzon comment, why should we deviate from the standard pattern? what is the benefit here? I'm fine doing it if there is a significant gain, otherwise it just adds an exception to what one would expect.